rcl/test/rcl/test_context.cpp

```diff
@@ -170,6 +174,10 @@ TEST_F(CLASSNAME(TestContextFixture, RMW_IMPLEMENTATION), bad_fini) {
   ret = rcl_shutdown(&context);
   EXPECT_EQ(ret, RCL_RET_OK);
 
-  ret = rcl_context_fini(&context);
```

Comment:
Does `inject_on_return` perform the same functionality as the non-mocked function and just return a new different value?

Reply:
Correct.